platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/events/performance/JankStatsMonitor.kt

```diff
@@ -0,0 +1,169 @@
+// capture-sdk - bitdrift's client SDK
+// Copyright Bitdrift, Inc. All rights reserved.
+//
+// Use of this source code is governed by a source available license that can be found in the
+// LICENSE file or at:
+// https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt
+
+package io.bitdrift.capture.events.performance
+
+import android.os.Build
+import android.view.Window
+import androidx.annotation.WorkerThread
+import androidx.metrics.performance.FrameData
+import androidx.metrics.performance.JankStats
+import io.bitdrift.capture.ErrorHandler
+import io.bitdrift.capture.LogLevel
+import io.bitdrift.capture.LogType
+import io.bitdrift.capture.LoggerImpl
+import io.bitdrift.capture.common.IBackgroundThreadHandler
+import io.bitdrift.capture.common.Runtime
+import io.bitdrift.capture.common.RuntimeFeature
+import io.bitdrift.capture.events.lifecycle.ILifecycleWindowListener
+import io.bitdrift.capture.events.span.SpanField
+import io.bitdrift.capture.providers.toFieldValue
+import io.bitdrift.capture.threading.CaptureDispatchers
+
+/**
+ * Reports Jank Frames and its duration in ms
+ *
+ * This will be a no-op when `client_feature.android.jank_stats_reporting` kill switch is disabled.
+ *
+ */
+internal class JankStatsMonitor(
+    private val logger: LoggerImpl,
+    private val runtime: Runtime,
+    private val errorHandler: ErrorHandler,
+    private val backgroundThreadHandler: IBackgroundThreadHandler = CaptureDispatchers.CommonBackground,
+) : ILifecycleWindowListener,
+    JankStats.OnFrameListener {
+    private var jankStats: JankStats? = null
+
+    override fun onWindowAvailable(window: Window) {
+        stopCollection()
+
+        if (runtime.isEnabled(RuntimeFeature.JANK_STATS_EVENTS)) {
+            setJankStatsForCurrentWindow(window)
+        }
+    }
+
+    override fun onWindowRemoved() {
+        stopCollection()
+    }
+
+    override fun onFrame(volatileFrameData: FrameData) {
+        if (!runtime.isEnabled(RuntimeFeature.JANK_STATS_EVENTS)) {
+            stopCollection()
+            return
+        }
+
+        if (volatileFrameData.isJank) {
+            // Below API 24 [onFrame(volatileFrameData)] call happens on the main thread
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.N) {
+                backgroundThreadHandler.runAsync { volatileFrameData.sendJankFrameData() }
+            } else {
+                // For >= 24 this happens on `FrameMetricsAggregator` thread
+                volatileFrameData.sendJankFrameData()
+            }
+        }
+    }
+
+    private fun setJankStatsForCurrentWindow(window: Window) {
+        try {
+            jankStats = JankStats.createAndTrack(window, this)
+
+            // BIT-4665 To update Runtime to provide heuristics value from config
```

Comment:
can you mark these with `// TODO(FranAguilera): ...` that's the current format we follow (plus allows you to list all the todos in android studio)